scripts/autoscaling/azure/nodeup.py

```diff
@@ -388,15 +410,23 @@ def create_vm(instance_name, run_id, instance_type, instance_image, disk, user_d
             }
         }
 
-    creation_result = compute_client.virtual_machines.create_or_update(
-        resource_group_name,
-        instance_name,
-        vm_parameters
-    )
-    creation_result.result()
-
-    start_result = compute_client.virtual_machines.start(resource_group_name, instance_name)
-    start_result.wait()
+    try:
+        creation_result = compute_client.virtual_machines.create_or_update(
+            resource_group_name,
+            instance_name,
+            vm_parameters
+        )
+        creation_result.result()
+
+        start_result = compute_client.virtual_machines.start(resource_group_name, instance_name)
+        start_result.wait()
+    except CloudError as client_error:
+        error_message = client_error.__str__()
+        if 'OperationNotAllowed' in error_message or 'ResourceQuotaExceeded' in error_message:
```

Comment:
Maybe we should check for `AND` condition instead of `OR` in an error message , I'm not sure whether  `OperationNotAllowed` will always refer to quotas problem?

Reply:
According to azure [docs](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-common-deployment-errors) `OperationNotAllowed` error code means:
 `The deployment is attempting an operation that exceeds the quota for the subscription, resource group, or region. If possible, revise your deployment to stay within the quotas. Otherwise, consider requesting a change to your quotas.`
This way I suppose the problem will always connected with quota. 
`ResourceQuotaExceeded` error code means the same and according to this [article](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-quota-errors) we can face it too.